src/Storages/ObjectStorage/StorageObjectStorageSettings.cpp

```diff
@@ -0,0 +1,104 @@
+#include <Core/BaseSettings.h>
+#include <Core/BaseSettingsFwdMacrosImpl.h>
+#include <Parsers/ASTCreateQuery.h>
+#include <Parsers/ASTFunction.h>
+#include <Parsers/ASTSetQuery.h>
+#include <Storages/ObjectStorage/StorageObjectStorage.h>
+#include <Storages/ObjectStorage/StorageObjectStorageSettings.h>
+#include <Storages/System/MutableColumnsAndConstraints.h>
+#include <Common/Exception.h>
+
+namespace DB
+{
+
+#define STORAGE_OBJECT_STORAGE_RELATED_SETTINGS(DECLARE, ALIAS) DECLARE(Bool, allow_dynamic_metadata_for_data_lakes, false, "", 0)
```

Comment:
What was the reason to make it a storage setting and not query-level setting?
The benefit from query-level setting is that it can be changed by `compatibility` setting. So we can set it to `true` by default and it will be disabled for all services with `compatibility < 24.N`

Reply:
For me it is difficult to imagine how this setting can be applied on a query level. It marks either that the CH table supports loading dynamic schema refreshing or behaves as an ordinary table (without changing schema externally)

Reply:
Negotiated that for now we introduce a table-level setting with false default value